<?xml version="1.0" ?>
<launch>
  <!-- Launches the nodes required for the 3D recognition pipeline. -->

  <arg name="weights_filename" default="/root/o2ac-ur/catkin_ws/src/o2ac_vision/config/cooking2022-11/weights/best.pt"/>

  <!-- Name of nodelet manager (without nodelet if empty string is given) -->
  <arg name="manager"			default=""/>

  <!-- AIST depth_filter parameters -->
  <arg name="camera_info_topic"	default="/camera_multiplexer/camera_info"/>
  <arg name="image_topic"	default="/camera_multiplexer/image"/>
  <arg name="depth_topic"	default="/camera_multiplexer/depth"/>
  <arg name="normal_topic"	default=""/>

  <!-- Spawn vision server -->
  <node name="o2ac_vision_server"
	pkg="o2ac_vision" type="o2ac_vision_server_cooking.py" output="screen">
    <remap from="/camera_info"		to="$(arg camera_info_topic)"/>
    <remap from="/image"		to="$(arg image_topic)"/>
    <remap from="/depth"		to="$(arg depth_topic)"/>
    <param name="weights_filename"		value="$(arg weights_filename)"/>
  </node>


</launch>
